v1

DenFedor opened the pull request:
HW2-express REST API backend for contacts.

DenFedor committed:
v1

HW2-express REST API backend for contacts.

DenFedor committed:
Added name,email validation patterns

Name : a-z,A-Z;
Email: should include @ and .

Harryk13 commented on models/contacts.js:
```diff
@@ -1,19 +1,84 @@
+    const contacts = await listContacts();
+    const index = contacts.findIndex((item) => item.id === contactId);
+    if (index !== -1) {
+      contacts[index].name = name;
```
що робити, якщо нам пришлють не усі проперті на зміну?

DenFedor replied:
Змінив обробку даних. Зараз динамічно береться із боді.

Harryk13 commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,86 @@
-})
+router.put("/:contactId", async (req, res, next) => {
+  try {
+    const { error } = contactsSchema.validate(req.body);
```
як мені оновити тільки одне проперті, а не усі одразу?

DenFedor replied:
Додав окрему схему валідації для пут запиту.

DenFedor committed:
Fixed update fucntion and validation schemas

Updated put method for contacts update. Now it doesn't require all fields for update.
Updated validation schemas.
Two body validation schemas being made.
One for add contact.
One for update contact.
